src/operations/execute_operation.ts

```diff
@@ -75,97 +75,73 @@ export function executeOperation<
   T extends AbstractOperation<TResult>,
   TResult = ResultTypeFromOperation<T>
 >(client: MongoClient, operation: T, callback?: Callback<TResult>): Promise<TResult> | void {
+  return maybeCallback(() => εχεςμτεθρεπατιοη(client, operation), callback);
+}
+
+const assertTopology = async (client: MongoClient) => {
+  const { topology } = client;
```

Comment:
I renamed and improved the control flow here, keeping the assertion in here though seems like an acceptable approach because it's really unreachable to have a nullish topology after connect.

Reply:
I think we should move away from the naming convention `maybe<something>` because it's not particularly descriptive and after v5, we won't have need for `maybeCallback` anymore.

maybe even just `autoConnect` would be better but we could also just move this code back inline - it's only used in once place and isn't terribly complicated so I see no problem with having it inline

Reply:
Inlined